nodebuilder/tests/blob_test.go

```diff
@@ -123,6 +123,37 @@ func TestBlobModule(t *testing.T) {
 				require.ErrorContains(t, err, blob.ErrBlobNotFound.Error())
 			},
 		},
+		{
+			name: "Submit equal blobs",
+			doFn: func(t *testing.T) {
+				appBlob, err := blobtest.GenerateV0Blobs([]int{8, 4}, true)
+				require.NoError(t, err)
+				b, err := blob.NewBlob(
+					appBlob[0].ShareVersion,
+					append([]byte{appBlob[0].NamespaceVersion}, appBlob[0].NamespaceID...),
+					appBlob[0].Data,
+				)
+				require.NoError(t, err)
+				height, err := fullClient.Blob.Submit(ctx, []*blob.Blob{b, b}, nil)
+				require.NoError(t, err)
+				_, err = fullClient.Header.WaitForHeight(ctx, height)
+				require.NoError(t, err)
+				b0, err := fullClient.Blob.Get(ctx, height, b.Namespace(), b.Commitment)
+				require.NoError(t, err)
+				require.Equal(t, b, b0)
+
+				// give some time to store the data,
+				// otherwise the test will hang on the IPLD level.
```

Comment:
Issue link

Reply:
#2915 
Hmmm. The deeper issue here seems to be that IPLD is hanging on the second time (if we give store long enough to store, it gets from StoreGetter)

Our ideas here are:
1. Bitswap peers maybe have an anti-spam mechanism where the same block will not be served twice to the same peer, which is fine, but:
2. These blocks should be cached on the blockstore level and not actually rerequested from the network. This makes it seem like the caching mechanism from IPLD retrieval isn't actually working.